tests/component/records/schema/field/test_pandas.py

```diff
@@ -0,0 +1,109 @@
+        # This test is designed to break when a new field type is
+        # introduced, so you can add new expectations and make sure
+        # the code handles the new type!
+
+        fields = {
+            'integer': {
+                'series': pd.Series([30, 35, 40]),
+                'constraints_type': RecordsSchemaFieldIntegerConstraints,
+                'statistics_type': type(None),
+            },
+            'decimal': {
+                'series': pd.Series([30.0, 35.1, 40.2]),
+                'constraints_type': RecordsSchemaFieldDecimalConstraints,
+                'statistics_type': type(None),
+            },
+            'string': {
+                'series': pd.Series(['a', 'b', 'c']),
+                'constraints_type': RecordsSchemaFieldStringConstraints,
+                'statistics_type': RecordsSchemaFieldStringStatistics,
+            },
+            'boolean': {
+                'series': pd.Series([True, True, False]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'date': {
+                'series': pd.Series([datetime.date(2020, 1, 1)]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'time': {
+                'series': pd.Series([datetime.time(hour=12, minute=0, second=0)]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'timetz': {
+                'series': pd.Series([datetime.time(hour=12, minute=0, second=0,
+                                                   tzinfo=pytz.timezone('US/Eastern'))]),
+                # refine_field_from_series() is not smart enough to
+                # distinguish whether the time objects inside it all
+                # have timezones or not.
+                'expected_field_type': 'time',
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'datetime': {
+                'series': pd.Series([datetime.datetime(2020, 1, 1, hour=12)]),
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            },
+            'datetimetz': {
+                'series': pd.Series([datetime.datetime(2020, 1, 1, hour=12,
+                                                       tzinfo=pytz.timezone('US/Eastern'))]),
+                # refine_field_from_series() is not smart enough to
+                # distinguish whether the datetime objects inside it all
+                # have timezones or not.
+                'expected_field_type': 'datetime',
+                'constraints_type': RecordsSchemaFieldConstraints,
+                'statistics_type': type(None),
+            }
+        }
+        for field_type in RECORDS_FIELD_TYPES:
```

Comment:
Pulling this list from the canonical Python type ensures we're testing each of the valid types.